### PR TITLE
[#266] Add functions for SHA cheksums for a file

### LINKED ISSUE #266: Security: Create functions for SHA checksums for a file
Create functions

```
int
pgmoneta_create_sha256_file(char* path, char** sha);

int
pgmoneta_create_sha384_file(char* path, char** sha);

int
pgmoneta_create_sha512_file(char* path, char** sha);
```

using the OpenSSL API

### PR DESCRIPTION
**Pull request by shikharish**
Closes #266

### CODE REVIEW

File: src/include/security.h [hunk 1]
```diff
@@ -89,14 +89,42 @@ pgmoneta_get_master_key(char** masterkey);
...
 
+/**
+ * Common function for generating hash for a file
+ * @param filename The file path
+ * @param hash The hash value
+ * @param md_name The md type(SHA256, SHA384, SHA512)
+ * @return 0 upon success, otherwise 1
+ */
+int
+pgmoneta_create_hash_file(char* filename, char** hash, const char* md);
```
Comment (jesperpedersen) [inline src/include/security.h:100]: I think this should be a `static` function inside `security.c` called `create_hash_file()`.

And, it should be `char* filename, char* algorithm, char** hash)` as the signature, first input then output after

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
Only a few minor changes found in my first pass - good job

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Comment by jesperpedersen** [comment]
@shikharish Do you want to tackle #291 next ? Its a lot harder...

**Comment by shikharish** [comment]
@jesperpedersen Yes, I will give it a try! If you can share any related resources, that would be really helpful.

**Comment by jesperpedersen** [comment]
Just make a comment on the issue that you will take it

**Pull request merged**